config/routes.rb

```diff
@@ -27,6 +27,7 @@
   resources :announcements
   resources :feeds, only: %i(index)
   resources :users, only: %i(index show new create destroy) do
+    resources :retirements, only: %i(index new), controller: "users/retirements"
```

Comment:
```suggestion
    resource :retirement, only: %i(index new), controller: "users/retirement"
```

複数のリタイアを扱うわけじゃないので単数形の方がいいと思います。